optimum/onnx/configuration.py

```diff
@@ -0,0 +1,229 @@
+        """
+        Mapping containing the axis definition of the output tensors(including loss) to provide to the model
+        Returns:
+            For each output: its name associated to the axes symbolic name and the axis position within the tensor
+        """
+        common_outputs = self._onnx_config.outputs
+        common_outputs.update(self._tasks_to_extra_outputs["default"])
+        if "loss" in common_outputs.keys():
+            common_outputs.move_to_end("loss", last=False)
+        return copy.deepcopy(common_outputs)
+
+    def generate_dummy_inputs(
+        self,
+        preprocessor: Union["PreTrainedTokenizerBase", "FeatureExtractionMixin"],
+        batch_size: int = -1,
+        seq_length: int = -1,
+        is_pair: bool = False,
+        framework: Optional[TensorType] = None,
+        num_channels: int = 3,
+        image_width: int = 40,
+        image_height: int = 40,
+        tokenizer: "PreTrainedTokenizerBase" = None,
+    ) -> Mapping[str, Any]:
+        """
+        Generate inputs to provide to the ONNX exporter for the specific framework
+        Args:
+            preprocessor: ([`PreTrainedTokenizerBase`] or [`FeatureExtractionMixin`]):
+                The preprocessor associated with this model configuration.
+            batch_size (`int`, *optional*, defaults to -1):
+                The batch size to export the model for (-1 means dynamic axis).
+            seq_length (`int`, *optional*, defaults to -1):
+                The sequence length to export the model for (-1 means dynamic axis).
+            is_pair (`bool`, *optional*, defaults to `False`):
+                Indicate if the input is a pair (sentence 1, sentence 2)
+            framework (`TensorType`, *optional*, defaults to `None`):
+                The framework (PyTorch or TensorFlow) that the tokenizer will generate tensors for.
+            num_channels (`int`, *optional*, defaults to 3):
+                The number of channels of the generated images.
+            image_width (`int`, *optional*, defaults to 40):
+                The width of the generated images.
+            image_height (`int`, *optional*, defaults to 40):
+                The height of the generated images.
+        Returns:
+            Mapping[str, Tensor] holding the kwargs to provide to the model's forward function
+        """
+        # Generate dummy labels
+        dummy_inputs = super().generate_dummy_inputs(
+            preprocessor,
+            batch_size,
+            seq_length,
+            is_pair,
+            framework,
+            num_channels,
+            image_width,
+            image_height,
+            tokenizer,
+        )
+        for label, input in self._tasks_to_extra_inputs[self.task].items():
+            if "sequence" in input.values():
+                dummy_inputs[label] = torch.zeros(
+                    self.default_fixed_batch, self.default_fixed_sequence, dtype=torch.long
```

Comment:
I think you should use `compute_effective_axis_dimension` to determine the batch size and the sequence length.

Reply:
Indeed I did not consider the case when sequence length and batch size are not dynamic. However, the labels unlike other elements in inputs don't need to consider the tokens to add, so it is either default_fixed_batch/seq(dynamic) or  batch_size/seq. I will use `compute_effective_axis_dimension` by setting `token_to_add=0`, thanks!